pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/UuidPreProcessor.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.pantheon.asciidoctor.extension;
+
+import org.asciidoctor.ast.Document;
+import org.asciidoctor.extension.Preprocessor;
+import org.asciidoctor.extension.PreprocessorReader;
+import com.redhat.pantheon.jcr.JcrQueryHelper;
+import org.apache.sling.api.resource.ResourceResolver;
+import java.util.Optional;
+
+import org.apache.sling.api.resource.Resource;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UuidPreProcessor extends Preprocessor {   // (1)
+
+    private static Resource module;
+    private static final Logger log = LoggerFactory.getLogger(UuidPreProcessor.class);
+    private static String newModulePath;
+
+    public UuidPreProcessor(Resource module) {
+        this.module = module;
+    }
+
+    @Override
+    public void process(Document document, PreprocessorReader reader) {
+        
+        List<String> lines = reader.readLines(); 
+        List<String> newLines = new ArrayList<String>();
+        String[] split;
+        String uuid, newLink;
+        
+        for (String line: lines) {
+            if(line.startsWith("xref:")){
+                split = line.split(",pantheon-id=");
+                uuid = split[1].replace(split[1].substring(split[1].length()-1), "");                
+                resolveActualPath(module.getResourceResolver(), uuid);
+                newLink = split[0].replaceAll(":.*?\\[", ":"+newModulePath+"[");                
```

Comment:
OOF, regex syntax!

I think I can read this. It mostly makes sense, but what is the purpose of the `?` in the expression?

Reply:
Explained the use of "?" in the below example:
input = "The << first match >> and << second match >> of the input string"
regex = <<.*>> matches the whole string "<< first match >> and << second match >>"
But
regex = <<.*?>> produces two matches: "<< first match >>" and "<< second match >>"

Reply:
Ahhhh, the "?" makes it non-greedy! Ok, that makes sense, thank you very much!